src/main/java/com/microsoft/lst_bench/client/JDBCConnection.java

```diff
@@ -75,12 +84,23 @@ private QueryResult execute(String sqlText, boolean ignoreResults) throws Client
                 + " retries) unsuccessful; stack trace: "
                 + ExceptionUtils.getStackTrace(e);
         if (errorCount == this.maxNumRetries) {
+          // Log any pending warnings associated with this statement, useful for debugging.
+          LOGGER.error(createSQLWarningMessage(s));
```

Comment:
I think that was the original point of the request since in SQL Server, the warnings contain statement information like the id. That information would not be logged in case of an exception because it would fail before then.

Reply:
Should we then print warnings for all statements, placing this line within the catch block, rather than restricting it to when `errorCount == this.maxNumRetries`? It seems more reasonable, considering that warnings are not necessarily errors. We should have a policy that makes sense for all systems, rather than solely relying on SQLServer behavior.

Reply:
Yes, makes completely sense to change it for each execution individually to allow track every exception call.